manifests/init.pp

```diff
@@ -54,25 +55,27 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  $rh_user                           = undef,
-  $rh_password                       = undef,
-  $org                               = undef,
-  $activationkey                     = undef,
-  $pool                              = undef,
-  $proxy_hostname                    = undef,
-  $proxy_port                        = undef,
-  $proxy_user                        = undef,
-  $proxy_password                    = undef,
-  $baseurl                           = 'https://cdn.redhat.com',
-  $servername                        = 'subscription.rhsm.redhat.com',
-  $serverprefix                      = '/subscription',
-  $serverport                        = 443,
-  $ca_cert_dir                       = '/etc/rhsm/ca/',
-  $repo_ca_cert_filename             = 'redhat-uep.pem',
-  $repo_ca_cert_source               = undef,
-  $manage_repos                      = 1,
-  $full_refresh_on_yum               = 0,
-  $package_ensure                    = 'latest',
+  Optional[String] $rh_user = undef,
+  Optional[String] $rh_password = undef,
+  Optional[String] $org = undef,
+  Optional[String] $activationkey = undef,
+  Optional[String] $pool = undef,
+  Optional[Stdlib::Fqdn] $proxy_hostname = undef,
+  Optional[Stdlib::Port] $proxy_port = undef,
+  Optional[String] $proxy_user = undef,
+  Optional[String] $proxy_password = undef,
+  Optional[Variant[Stdlib::Filesource,Pattern[/\Apuppet:\/\/.+/]]] $repo_ca_cert_source = undef,
+  Integer[0] $server_timeout = 180,
+  Stdlib::Httpurl $baseurl = 'https://cdn.redhat.com',
+  Stdlib::Fqdn $servername = 'subscription.rhsm.redhat.com',
+  Stdlib::Absolutepath $serverprefix = '/subscription',
+  Stdlib::Port $serverport = 443,
+  Stdlib::Absolutepath $ca_cert_dir = '/etc/rhsm/ca/',
+  String $repo_ca_cert_filename = 'redhat-uep.pem',
+  Integer[0,1] $manage_repos = 1,
+  Integer[0,1] $full_refresh_on_yum = 0,
+  Integer[0,1] $inotify = 1,
+  Enum['present','absent','held','latest','installed','purged'] $package_ensure = 'latest',
```

Comment:
Maybe you don't want to be so restrictive here and change it to `String`, since one could also want to specify a specific version.
According to `puppet describe package`, values can match `/./`.

Reply:
Good catch, changed to `String[1]` and extended the param description.